runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5061,13 +5061,57 @@ TR_J9ByteCodeIlGenerator::loadAuto(TR::DataType type, int32_t slot, bool isAdjun
    push(load);
    }
 
+static bool isFieldResovled(TR::Compilation *comp, TR_ResolvedJ9Method * owningMethod, int32_t cpIndex, bool isStore)
+   {
+   uint32_t offset = 0;
+   TR::DataType type = TR::NoType;
+   bool isVolatile = true, isFinal = false, isPrivate = false, isUnresolvedInCP;
+   bool bResolved = owningMethod->fieldAttributes(comp, cpIndex, &offset, &type, &isVolatile, &isFinal,
```

Comment:
What happens if the cpIndex is not field entry in the CP?

Reply:
Depending on which one gets to check first (https://github.com/eclipse/openj9/pull/10362#discussion_r473185338): checking `isFieldQType()` first or checking `isFieldResovled()` first. 
- If  `isFieldQType()` is called first, it returns false when `cpIndex == -1`. It goes down the existing path of `loadInstance(symRef)`. It looks to me it still creates a symRef even when `cpIndex == -1` regardless of value class or not. 
- If `isFieldResovled()` is called first, it relies on `owningMethod->fieldAttributes` to check cpIndex. `TR_ResolvedJ9Method::fieldAttributes()` only has an `TR_ASSERT(cpIndex != -1, ...)`. It means the release build could crash when `cpIndex == -1`. However it applies to all cases when fieldAttributes is called regardless of value classes or not.